db/attachment.go

```diff
@@ -37,8 +37,15 @@ var (
 	ErrAttachmentMeta = base.HTTPErrorf(http.StatusBadRequest, "Invalid _attachments")
 )
 
-// AttachmentData holds the attachment key and value bytes.
-type AttachmentData map[string][]byte
+// updatedAttachment represents an attachment that has been modified by a create or update operation.
+type updatedAttachment struct {
+	body    []byte // The attachment data in bytes
+	updated bool   // If this attachment was an update of an existing attachments
+	created bool   // If this attachment was new on the body
```

Comment:
Given that it's always only one of `updated` or `created`, it feels like two flags aren't necessary?